code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -156,103 +273,49 @@
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
 	///Adds a cooldown to deactivation to avoid accidental cancel
 	COOLDOWN_DECLARE(misclick_prevention)
 
 /datum/action/xeno_action/activable/feast/can_use_ability(atom/target, silent, override_flags)
 	. = ..()
-	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/carbon/xenomorph/owner_xeno = owner
 
 	if(!COOLDOWN_CHECK(src, misclick_prevention))
 		return FALSE
-	if(X.has_status_effect(STATUS_EFFECT_XENO_FEAST))
+	if(owner_xeno.has_status_effect(STATUS_EFFECT_XENO_FEAST))
 		return TRUE
 
-/datum/action/xeno_action/activable/feast/use_ability()
-	var/mob/living/carbon/xenomorph/X = owner
-	if(X.has_status_effect(STATUS_EFFECT_XENO_FEAST))
-		to_chat(X, span_notice("We decide to end our feast early..."))
-		X.remove_status_effect(STATUS_EFFECT_XENO_FEAST)
+/datum/action/xeno_action/activable/feast/use_ability(atom/A)
+	. = ..()
+	var/mob/living/carbon/xenomorph/owner_xeno = owner
+	if(owner_xeno.has_status_effect(STATUS_EFFECT_XENO_FEAST))
+		to_chat(owner_xeno, span_notice("We decide to end our feast early..."))
+		owner_xeno.remove_status_effect(STATUS_EFFECT_XENO_FEAST)
 		return
 
-	X.emote("roar")
-	X.visible_message(X, span_notice("[X] begins to overflow with vitality!"))
-	X.apply_status_effect(/datum/status_effect/xeno_feast, GORGER_FEAST_DURATION, X.xeno_caste.feast_plasma_drain)
+	owner_xeno.emote("roar")
+	owner_xeno.visible_message(owner_xeno, span_notice("[owner_xeno] begins to overflow with vitality!"))
+	owner_xeno.apply_status_effect(/datum/status_effect/xeno_feast, GORGER_FEAST_DURATION, owner_xeno.xeno_caste.feast_plasma_drain)
 	COOLDOWN_START(src, misclick_prevention, 2 SECONDS)
-	succeed_activate()
 	add_cooldown()
+	succeed_activate()
 
-/////////////////////////////////
-// Devour
-/////////////////////////////////
-/datum/action/xeno_action/activable/devour
-	name = "Devour"
-	action_icon_state = "regurgitate"
-	mechanics_text = "Devour your victim to be able to carry it faster."
-	use_state_flags = XACT_USE_STAGGERED|XACT_USE_FORTIFIED|XACT_USE_CRESTED //can't use while staggered, defender fortified or crest down
-	keybind_signal = COMSIG_XENOABILITY_REGURGITATE
-	plasma_cost = 0
-	target_flags = XABB_MOB_TARGET
-
-/datum/action/xeno_action/activable/devour/can_use_ability(atom/A, silent, override_flags)
-	. = ..()
-	if(!.)
-		return
-	var/mob/living/carbon/xenomorph/X = owner
-	if(X.eaten_mob)
-		return TRUE
-	if(!ishuman(A) || issynth(A))
-		if(!silent)
-			to_chat(owner, span_warning("That wouldn't taste very good."))
-		return FALSE
-	var/mob/living/carbon/human/victim = A
-	if(owner.do_actions) //can't use if busy
-		return FALSE
-	if(!owner.Adjacent(victim)) //checks if owner next to target
-		return FALSE
-	if(!HAS_TRAIT(victim, TRAIT_UNDEFIBBABLE))
-		if(!silent)
-			to_chat(owner, span_warning("This creature is struggling too much for us to devour it."))
+/datum/action/xeno_action/activable/feast/ai_should_use(atom/target)
```

Comment:
check if target is atom AND not a friendly xeno

Reply:
This ability doesn't care about target, why should I check it?